CMakeLists.txt

```diff
@@ -14,6 +14,7 @@ set(ABI_VERSION 0)
 # Options
 option(ENABLE_TESTS "Enable all tests and checks" OFF)
 option(ENABLE_COVERAGE "Enable coverage reports (includes enabling all tests and checks)" OFF)
+option(ENABLE_WERROR "Treat all build warnings as errors" OFF)
```

Comment:
If we disable it by default, we should inject -DENABLE_WERROR=ON in .build.yml, so that your CI builds always use -Werror.